export/export.py

```diff
@@ -0,0 +1,866 @@
+            self._run_pipeline("source_transform")
+        # Otherwise start with quantize stage if available
+        elif "quantize" in self._stages:
+            self._run_pipeline("quantize")
+        # Otherwise start with export stage
+        else:
+            self._run_pipeline("export")
+
+    def save_pte_file(self, path: str) -> None:
+        """
+        Save the exported program to a PTE file.
+
+        Args:
+            path: Path where the PTE file will be saved
+
+        Raises:
+            RuntimeError: If the executorch program manager is not initialized
+        """
+        if self._executorch_program_manager is None:
+            raise RuntimeError(
+                "Executorch program manager is not initialized. Run export() first."
+            )
+        self._executorch_program_manager.save(path)
+
+    def get_executorch_program(self) -> Program:
+        """
+        Get the ExecutorchProgram from the ExecutorchProgramManager.
+
+        Returns:
+            The ExecutorchProgram
+
+        Raises:
+            RuntimeError: If the executorch program manager is not initialized
+        """
+        if self._executorch_program_manager is None:
+            raise RuntimeError(
+                "Executorch program manager is not initialized. Run export() first."
+            )
+        return self._executorch_program_manager.executorch_program
+
+    def get_pte_buffer(self) -> bytes:
+        """
+        Get the PTE buffer as bytes.
+
+        Returns:
+            The PTE buffer as bytes
+
+        Raises:
+            RuntimeError: If the executorch program manager is not initialized
+        """
+        if self._executorch_program_manager is None:
+            raise RuntimeError(
+                "Executorch program manager is not initialized. Run export() first."
+            )
+        return self._executorch_program_manager.buffer
+
+    def get_example_input(
+        self, method_name: str = "forward"
+    ) -> Tuple[torch.Tensor, ...]:
+        """
+        Get the example input for a specific method.
+
+        Args:
+            method_name: Name of the method to get example input for, defaults to "forward"
+
+        Returns:
+            Tuple of tensors representing the example input
+
+        Raises:
+            KeyError: If the method name is not found in example inputs
+            ValueError: If the example inputs list is empty
+        """
+        if method_name not in self._example_inputs:
+            raise KeyError(f"Method name '{method_name}' not found in example inputs")
+
+        # Access the first element of the list for this method
+        example_inputs_list = self._example_inputs[method_name]
+        if not example_inputs_list:
+            raise ValueError(f"Example inputs list for method {method_name} is empty")
+
+        # The original code expects this to be a tuple of tensors
+        return self._example_inputs[method_name][0]
+
+    def run_method(
```

Comment:
why is this part of this class

Reply:
I think that's a valid point, but i've felt like having all the quantization configuration related items in one class would generally be easier for recipe authors. We when creating the pipeline decide what source transforms need to be run and what is in pt2e. This isn't set in stone though, we'll definitely iterate on this as we add more recipes in XNNPack, CoreML etc. and we can arrive at a different structure over the next few weeks if the current one isn't flexible enough.